### DISCUSSION TITLE
Allow async transform docs.source.transform

### DISCUSSION THREAD

**Discussion by SimmeNilsson** [storybookjs/storybook#23703] [Ideas]
### Is your feature request related to a problem? Please describe.

Prettier has recently switch to an async API.
We are using prettier format in docs.source.transform.
https://storybook.js.org/docs/html/api/doc-block-source#transform

### Describe the solution you'd like

Would it be possible to allow for a Promise<string> to be returned from transform?

### Describe alternatives you've considered

Tried @prettier/sync but it is dependent on node worker_threads. Breaks storybook build.

### Are you able to assist to bring the feature to reality?

no

### Additional context

Example of usage, but format is async in prettier 3.x.
`transform: (input: string) => {
    return prettier.format(input, {
        parser: "html",
        plugins: [prettierHtml]
    });
}`

**Comment by omarkawach** [comment]
Also interested in this

**Comment by vprothais** [comment]
The [HTML preview addon](https://storybook.js.org/addons/@whitespace/storybook-addon-html/) needs Prettier 3 to format source in Story view, but we can't use Prettier 3 if we use prettier in `docs.source.transform`.

**Comment by valentinpalkovic** [comment]
Fixed by https://github.com/storybookjs/storybook/pull/30426
